MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -616,12 +644,12 @@ private static void setScenarioMapSize(AtBDynamicScenario scenario) {
     public static void setScenarioModifiers(AtBDynamicScenario scenario) {
         // this is hardcoded for now, but the eventual plan is to let the user configure how many modifiers
         // they want applied
-        int numMods = 2;//Compute.randomInt(2);
+        int numMods = Compute.randomInt(2);
         
         for(int x = 0; x < numMods; x++) {
             int scenarioIndex = Compute.randomInt(AtBScenarioModifier.getScenarioModifiers().size());
             
-            AtBScenarioModifier scenarioMod = AtBScenarioModifier.getScenarioModifiers().get(scenarioIndex);
+            AtBScenarioModifier scenarioMod = (AtBScenarioModifier) AtBScenarioModifier.getScenarioModifiers().values().toArray()[scenarioIndex];
```

Comment:
Well that's "fun", I realize `numMods` isn't going to be large, but this feels off. Probably best to hoist the array outside of the loop.

Reply:
Refactored with more intelligent implementation.